lizard_connector/connector.py

```diff
@@ -27,34 +31,52 @@
 
 import lizard_connector.queries
 
+ASYNC_POLL_TIME = 1
+ASYNC_POLL_TIME_INCREASE = 1.5
+
 
 class LizardApiTooManyResults(Exception):
     pass
 
 
+class LizardApiAsyncTaskFailure(Exception):
+    pass
+
+
 class InvalidUrlError(Exception):
     pass
 
 
+def no_op(*args, **kwargs):
+    pass
+
+
+def save_to_json(result, file_base="api_result"):
+    """
+    Saves a result to json with a timestamp in milliseconds.
+
+    Args:
+        result (list|dict): a json dumpable object to save to file.
+        file_base (str): filename base. Can contain a relative or absolute
+                         path.
+    """
+    filename = "{}_{}.json".format(file_base, str(int(time.time() * 1000)))
+    with open(filename, 'w') as json_filehandler:
+        json.dump(result, json_filehandler)
+
+
 class Connector(object):
 
-    def __init__(self, max_results=1000, username=None, password=None,
-                 all_pages=True):
+    def __init__(self, max_results=1000, username=None, password=None):
```

Comment:
max_results=1000

This seems unintended.
Pagination seems to work, but if there are more than 1000 assets to download, you get an error.

```
Traceback (most recent call last):
  File "/usr/lib/python3.5/code.py", line 91, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
  File "/home/madeleine.vanwinkel/code/lizard-connector/lizard_connector/connector.py", line 228, in __next__
    return self._next_page()
  File "/home/madeleine.vanwinkel/code/lizard-connector/lizard_connector/connector.py", line 203, in _next_page
    self._count = self._endpoint.handle_count(json_)
  File "/home/madeleine.vanwinkel/code/lizard-connector/lizard_connector/connector.py", line 103, in handle_count
    count or 0, self.max_results)
lizard_connector.connector.LizardApiTooManyResults: Too many results: 3259 found, while max 1000 are accepted
```